Content.Client/Smoking/BurnStateVisualizerSystem.cs

```diff
@@ -0,0 +1,42 @@
+using Robust.Client.GameObjects;
+using Content.Shared.Smoking;
+
+namespace Content.Client.Smoking;
+
+public sealed class BurnStateVisualizerSystem : VisualizerSystem<BurnStateVisualsComponent>
+{
+
+    [Dependency] private readonly IEntityManager _entMan = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<BurnStateVisualsComponent, ComponentInit>(OnComponentInit);
+    }
+
+    private void OnComponentInit(EntityUid uid, BurnStateVisualsComponent component, ComponentInit args)
+    {
+        IoCManager.InjectDependencies(this);
```

Comment:
definitely don't need this at all, entity system type classes already do this for you

Reply:
Erased